packages/vscode-extension/src/commonlib/telemetry.ts

```diff
@@ -92,8 +92,10 @@ export class VSCodeTelemetryReporter extends vscode.Disposable implements Teleme
       properties = { ...this.sharedProperties, ...properties };
     }
 
-    const projectId = getProjectId();
-    properties[TelemetryProperty.ProjectId] = projectId ? projectId : "";
+    if (properties[TelemetryProperty.ProjectId] === "unknown") {
```

Comment:
Is there telemetry event that are sent before you set 'unknown' to project-id? In this situation the property will not be attached(because the projectId property is undefined in this line), right?

Reply:
The project-id is set to sharedProperties in handlers.ts:activate() using `ExtTelemetry.addSharedProperty(TelemetryProperty.ProjectId, projectId);` It's before any telemetry events so ideally every telemetry event has this property except it's not found in a teamsfx project. This line of code is to try to get project id again that should not happen in normal situation.

Reply:
Yes, you're right. But I'm concerned if anyone adds telemetry event before handlers.activate(). In that case, the property-id will not be attached. Can we maybe use the statement `if (properties[TelemetryProperty.ProjectId] === "unknown" || properties[TelemetryProperty.ProjectId] === undefined)`  to avoid such condition?

Reply:
Thank you! That's right and I've fixed it.